lib/domain/entities/message/sms.ts

```diff
@@ -0,0 +1,143 @@
+import { Message, MessageData } from './message'
+import { normalizeDiacritics } from 'normalize-text'
+
+type MessageType = {
+  prefix?: string;
+  text: string;
+  suffix?: string;
+  variables?: Record<string, string>;
+};
+
+type RecipientType = {
+  phone: string;
+};
+
+export type SMSData = {
+  message: MessageType;
+  recipient: RecipientType;
+} & MessageData;
+
+type SMSOptions = {
+  shortify?: boolean;
+  char?: number;
+};
+
+export class SMS implements Message {
+  readonly channel: string = 'sms'
+  readonly externalId?: string
+  readonly recipient: RecipientType
+  private readonly message: MessageType
+  private readonly RESERVED_SPACE_FORMAT = 3
+  private readonly SEE_MORE = '... Veja mais em:'
+
+  constructor(
+    { message, recipient, externalId }: Pick<SMSData, 'message' | 'recipient' | 'externalId'>,
+    options?: SMSOptions
+  ) {
+    this.externalId = externalId
+    this.message = this.normalize(message)
+    this.recipient = recipient
+    this.replaceVariables()
+
+    if (options?.shortify) {
+      this.shortify(options.char)
+    }
+
+    this.build()
```

Comment:
A quem for olhar a PR, o que acham dessa abordagem? 

Motivo: A intenção em usar isso é mandar a mensagem pronta pro COM-Service

Se mantivesse chamando `message.shortify()` quando não quisesse usar o `shortify` mas passando prefix e suffix, teriamos duas opção:

1- mandaria com o suffix e o prefix e o COM-service montaria a mensagem final ou mandaria pro serviço de sms fazer isso.
2 - export o metodo build pra quem usar ter que chamar antes de fazer o dispatch mas eu particularmente não gostei pq ficaria algo assim:

```typescript 
const message = new SMS()

message.shortify()
message.build()

client.dispatch(message)
```

nessa forma que esta o COM-client ficaria responsavel por fazer isso automatico, usando ou não usando o shortify, montando a mensagem final do jeito esperado, seja com prefix e suffix, sem os dois ou sem um.

Reply:
Cara.. o get message não poderia chamar o `build()`?

dai ficaria só o 
```ts
const message = new SMS() 
message.shortify()
client.dispatch(message) // chama o build quando usar o getMessage
```